src/main/java/appeng/parts/reporting/PartConversionMonitor.java

```diff
@@ -89,62 +91,133 @@ public boolean onPartShiftActivate( final EntityPlayer player, final EnumHand ha
 			return false;
 		}
 
-		boolean ModeB = false;
+		final ItemStack eq = player.getHeldItem( hand );
+		if( this.isLocked() )
+		{
+			if( eq.isEmpty() )
+			{
+				this.insertItem( player, hand, true );
+			}
+			else if( Platform.isWrench( player, eq, this.getLocation().getPos() ) && ( this.getDisplayed() == null || !this.getDisplayed().equals( eq ) ) )
+			{
+				// wrench it
+				return super.onPartActivate( player, hand, pos );
+			}
+			else
+			{
+				this.insertItem( player, hand, false );
+			}
+		}
+		else if( this.getDisplayed() != null && this.getDisplayed().equals( eq ) )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this call to "equals"; comparisons between unrelated types always return false. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2159)